### PR TITLE
Update build workflows to allow manual runs 

### PR DESCRIPTION
**Pull request by sk593**
# Description

This updates the build workflow to allow manual runs for testing feature branch changes

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 name: Build and Test
 on:
+  # Enable manual trigger
+  workflow_dispatch:
```
Comment (youngbupark) [inline .github/workflows/build.yaml:20]: You might want to get the input for your feature branch. Please see this - https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/

Comment (sk593) [reply .github/workflows/build.yaml:20]: Where would the inputs be used for this? From my understanding, the manual run would use the workflow version from the feature branch that has the needed changes. Unless I'm missing something

Comment (youngbupark) [reply .github/workflows/build.yaml:20]: So do you just want to run build action manually from main branch? Or do you want to specify which feature branch you want to build ?

Comment (sk593) [reply .github/workflows/build.yaml:20]: Yes to the second question. I want to be able to run the workflow from the feature branch. So with the `workflow_dispatch` I can specify which branch to build the workflow off of

Comment (youngbupark) [reply .github/workflows/build.yaml:20]: image

It fetches the branches of the upstream repo. Is this expected? don't you want to build the branch in the forked repo?

**Review by youngbupark — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (youngbupark) [reply .github/workflows/build.yaml:20]: I just approved this PR. but you might need additional input if you want to support branches in forked repo.

Comment (sk593) [reply .github/workflows/build.yaml:20]: Ah no, so the feature branch is in the main repo not in a forked repo. So the branches of the upstream repo is correct

File: .github/workflows/functional-test.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ permissions:
   packages: write # Required for uploading the package
 
 on:
+  # Enable manual trigger
+  workflow_dispatch:
```
Comment (willdavsmith) [inline .github/workflows/functional-test.yaml:26]: if run via workflow dispatch, the tests might not work. but we should try it.

**Review by willdavsmith — APPROVED** [review]

**Pull request merged**